### ISSUE TITLE
Unit IsStartingAttack always returns false

### ISSUE THREAD

**Issue by Nerfling** [bwapi/bwapi#647]
Hello,

first of all, thanks for this awesome environment, I had lot of fun with it so far.

unfortunately I came upon which I think is a bug.

I'm trying to check `IsStartingAttack()` to see when my units attack but this always returns 0 (false)

I tried:
~~~cpp
for each (Unit u in Broodwar->self()->getunits())
     if (u->isStartingAttack())
         Broodwar->printf("attacking");
~~~
I also tried `(IsAttackFrame() & IsAttacking())` & tried using `UnitInterface` instead of `Unit`

I must be doing something wrong, I would be grateful if someone can help me out :)

Thanks in Advance,
Savant.

**Comment by heinermann** [comment]
Is it the latest version? Can someone verify?

**Comment by Nerfling** [comment]
Hello,

It is version 4.1.2.0, I'm testing 8 mutalisks on a training map against 20 marines, using map settings.

**Comment by Nerfling** [comment]
Update:

I tested this with just the ExampleAIModule and it seems to work there everytime, so the problem is probably somewhere in my other code, you can close this.

Sorry for the problems caused,
Savant.

**Comment by Nerfling** [comment]
Another update:

The problem seems to be caused when I had my bot make too many actions per minute. I checked for (Unit->IsStartingAttack()) after I assigned which action they should take, by limiting these actions (attack or move) by:

if (Broodwar->getFrameCount() % 5 == 0)
{
   //do actions
}

and then checking IsStartingAttack() it seems to work every time :)

I hope this can help some other people.

Thanks,
Savant.
